common/persistence/cassandra/cassandraPersistenceUtil.go

```diff
@@ -60,10 +60,6 @@ func applyWorkflowMutationBatch(
 	condition := workflowMutation.Condition
 
 	startVersion := workflowMutation.StartVersion
-	lastWriteVersion := workflowMutation.LastWriteVersion
-	// TODO remove once 2DC is deprecated
-	//  since current version is only used by 2DC
-	currentVersion := lastWriteVersion
```

Comment:
currentVersion should be deleted
startVersion can be deleted
lastWriteVersion cannot be deleted, this is used in multiple place, e.g. start workflow execution

Reply:
actually, the start version / last write version / current version can all be deleted here, since this is modifying  the concrete mutable state record.

there is another function, `createOrUpdateCurrentExecution`, you can remove the start version from that function.
the 2 lines to be removed:
https://github.com/temporalio/temporal/blob/36525fad464757f59d1e963ccba3b3c46aff0e33/common/persistence/sql/sqlExecutionManagerUtil.go#L574
https://github.com/temporalio/temporal/blob/36525fad464757f59d1e963ccba3b3c46aff0e33/common/persistence/cassandra/cassandraPersistenceUtil.go#L964

Reply:
Filed #702